Classes/Extensions/Amazing+Tools.swift

```diff
@@ -9,198 +9,190 @@
     }
     
 }
 
-
-extension Awesome.brand {
-
-    public typealias AwesomeType = Awesome.brand
-
-    public static func fromType(_ code: String) -> AwesomeType? {
-        guard let icon = AwesomeType(rawValue: code) else {
-            return nil
-        }
-        return icon
-    }
-
-    public var name: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.keys[index]
-        return name
+extension Amazing {
+    
+    /// Returns an unique identifier string that contains the font name, font style and icon name
+    public var detailedKey: String {
+        return String(reflecting: self)
+            .split(separator: ".")
+            .suffix(3)
+            .map { String($0).camelCaseToString(with: "-").lowercased() }
+            .joined(separator: ".")
     }
-
-    public var label: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.labels[index]
-        return name
+    
+    /// Returns an unique identifier string that contains the icon name
+    public var key: String {
+        return String(describing: self).lowercased()
     }
-
-}
-
-extension Awesome.regular {
-
-    public typealias AwesomeType = Awesome.regular
-
-    public static func fromType(_ code: String) -> AwesomeType? {
-        guard let icon = AwesomeType(rawValue: code) else {
-            return nil
-        }
-        return icon
+    
+    /// Returns a human readable string that describes the icon
+    public var description: String {
+        return String(describing: self).camelCaseToString()
     }
-
-    public var name: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.keys[index]
-        return name
+    
+    /// An array with all keys of all icons of the font style
+    public static var allKeys: [String] {
+        return allCases.map { $0.key }
     }
 
-    public var label: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.labels[index]
-        return name
+    /// An array with all detailed keys of all icons of the font style
+    public static var allDetailedKeys: [String] {
+        return allCases.map { $0.detailedKey }
     }
-
-}
-
-extension Awesome.solid {
-
-    public typealias AwesomeType = Awesome.solid
-
-    public static func fromType(_ code: String) -> AwesomeType? {
-        guard let icon = AwesomeType(rawValue: code) else {
-            return nil
-        }
-        return icon
+    
+    /// An array with all human readable descriptions of all icons of the font style
+    public static var allDescriptions: [String] {
+        return allCases.map { $0.description }
     }
-
-    public var name: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.keys[index]
-        return name
+    
+    @available(swift, deprecated: 4.2, message: "This will be removed in the future. Please use .allKeys instead")
```

Comment:
You can also do `@available(*, unavailable, renamed: "allKeys")` which will offer user the "red dot of fix" in the code ... this is how apple generates their "Fix it!" warnings ...

Reply:
I would not worry about compatibility as this will be a major update `2.0.0`

Reply:
That's a bit problematic as the computed property expects a return type that `fatalError` cannot meet. Addind `fatalError` before the `return` with the correct type generates compiler warnings that the return will never be executed. IMHO there is no need for the fatalError as the compiler will fail on the `unavailable` anyway. I'll have another look and submit a proposal.

Reply:
have you tried? :) ... you need to put it in the `get` method, set can be let empty ...

```swift
var prop: String {
   get {
      fatalError()
   }
   set { }
}
```

Reply:
Nice! I it works when using get/set but not when the properties are declared as read-only computed properties.